engine/function/src/main/java/io/deephaven/function/Logic.java

```diff
@@ -86,11 +85,10 @@ static public Boolean and(Boolean[] values, Boolean nullValue) {
      * @return logical and of all the values in the array. By convention, returns true if the array is empty.
      */
     static public Boolean and(ObjectVector<Boolean> values, Boolean nullValue) {
-        for (int ii = 0; ii < values.size(); ++ii) {
-            Boolean b = values.get(ii);
-            b = b == null ? nullValue : b;
+        for (final Boolean b : values) {
```

Comment:
We should consider whether we should check the empty case before making the iterator, and apply whatever convention we decide on universally. Some of the iterators are a little expensive to make, although we usually make cheaper ones for small/empty vectors. Even the cheapest iterators are still an allocation, although hopefully "escaped". This applies to a lot of methods, so just commenting once for discussion.

Reply:
We decided not to check `isEmpty()` unless we have a reason to in the function.